### PR TITLE
doc: Update list of valid PR areas

### PR DESCRIPTION
**Pull request by ch4ot1c**
Updates list of Pull Request areas / prefixes in `CONTRIBUTING.md`:

- Adds `init`, `cli`, `depends`, `contrib`
- Removes `lib`
- Changes `script` to `scripts`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -79,19 +79,22 @@ about Git.
 The title of the pull request should be prefixed by the component or area that
 the pull request affects. Valid areas as:
 
-  - `consensus` for changes to consensus critical code
```
Comment (MarcoFalke) [inline CONTRIBUTING.md]: why is this (re)moved?

**Review by jonatack — COMMENTED** [review]
I don't see the value of these changes.

**Comment by hebasto** [comment]
>     * Changes `script` to `scripts`

See: https://github.com/bitcoin/bitcoin/pull/16484#issuecomment-516095516

**Comment by laanwj** [comment]
> I don't see the value of these changes.

I agree. I'm also not sure this is worth doing.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ch4ot1c) [reply CONTRIBUTING.md]: Figured they should be ordered by simplicity.

**Comment by ch4ot1c** [comment]
Alright - I thought these would be validated by `commit-script-check` at some point; I guess these are relaxed instructions.

**Pull request closed without merging**